File: DotNetMessaging/SpringBootMessaging/src/main/java/com/messaging/api/models/Group.java
package com.messaging.api.models;

import lombok.Data;
import lombok.EqualsAndHashCode;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;

@Data
@EqualsAndHashCode(callSuper = true)
@Document(collection = "groups")
public class Group extends BaseEntity {
    private String name;
    
    private String description;
    
    private String profilePictureUrl;
    
    private String createdById;
    
    private LocalDateTime lastMessageAt;
}

File: DotNetMessaging/backend/DotNetMessaging.API/Models/Group.cs
using MongoDB.Bson.Serialization.Attributes;

namespace DotNetMessaging.API.Models;

public class Group : BaseEntity
{
    [BsonElement("name")]
    public string Name { get; set; } = string.Empty;

    [BsonElement("description")]
    public string? Description { get; set; }

    [BsonElement("profilePictureUrl")]
    public string? ProfilePictureUrl { get; set; }

    [BsonElement("createdById")]
    public string CreatedById { get; set; } = string.Empty;

    [BsonElement("lastMessageAt")]
    public DateTime? LastMessageAt { get; set; }
}
